src/polycubed/src/cube_tc.cpp

```diff
@@ -27,16 +27,71 @@ namespace polycubed {
 
 CubeTC::CubeTC(const std::string &name, const std::string &service_name,
                const std::vector<std::string> &ingress_code,
-               const std::vector<std::string> &egress_code, LogLevel level)
+               const std::vector<std::string> &egress_code, LogLevel level, bool shadow, bool span)
     : Cube(name, service_name, PatchPanel::get_tc_instance(),
-           PatchPanel::get_tc_instance(), level, CubeType::TC) {
+           PatchPanel::get_tc_instance(), level, CubeType::TC, shadow, span) {
   // it has to be done here becuase it needs the load, compile methods
   // to be ready
   Cube::init(ingress_code, egress_code);
+
+  if (shadow) {
+    auto res = ingress_programs_[0]->open_perf_buffer("span_slowpath", call_back_proxy, nullptr, this);
+    if (res.code() != 0) {
+      logger->error("cannot open perf ring buffer for span_slowpath: {0}", res.msg());
+      throw BPFError("cannot open span_slowpath perf buffer");
+    }
+    start();
+  }
+}
+
+void CubeTC::call_back_proxy(void *cb_cookie, void *data, int data_size) {
```

Comment:
1) What does "proxy" mean here?
2) Should we write "callback" or it is really "call_back"?